### ISSUE TITLE
PV63

### ISSUE THREAD

**Issue by bobsummerwill** [ethereum/cpp-ethereum#3102]
_From @chriseth on February 8, 2016 23:48_

This is implemented at the responding side but not at the asking side.
Explanation (implementation in go-ethereum): https://github.com/ethereum/go-ethereum/pull/1889

Description of older issue:

PV63 fastsync mode, launched if PV63 peers available && (start || a new block is noted with height > N + currentHeight)

1. Get last N block headers by hashes in ascending tree order. Pick block X = Latest - N 
2. In parallel:
 * Download state tree nodes for block X from PV63 peers
 * Download transaction receipts for blocks 1 to X from PV63 peers
 * Download 1..X block bodies and headers from PV62 and PV63 peers
3. Once done
 * Insert blocks
 * Insert state

PV62 sync mode, launched if no PV62 peers available && (start || a new block is noted with height > N + currentHeight)
2. Split chain into chunks of size M
Download chunks in parallel. For each chunk:
 * Download block headers in ascending tree order until a known block hash is met
 * Download block bodies
3. Reorder blocks and insert them into block queue

Keep up mode, launched when a new block is noted with height < N + currentHeight
* Download headers from a single peer in ascending tree order until a known block hash is met
* Download bodies

General:

* Validate all attack vectors, review code audit report.

_Copied from original issue: ethereum/webthree-umbrella#177_

**Comment by axic** [comment]
@gumb0 I think this must be implemented otherwise it wouldn't be able to sync with latest clients

**Comment by gumb0** [comment]
@axic No, this is fast sync description, still not implemented, we sync only through full sync (supported by all full clients)

But I haven't seen this issue, thanks.
